base/reduce.jl

```diff
@@ -133,7 +141,7 @@ function mapfoldr(f, op, itr)
     if isempty(itr)
```

Comment:
Unrelated comment, which can be done as part of this PR or in a later one: Since we have https://github.com/JuliaLang/julia/pull/24187 `Iterators.Reverse` now, I think we can write these `mapfoldr` functions generically. As it stands it only works with arrays, and this generalization may allow it to work with arbitrary reversible iterators.